src/hmac.rs

```diff
@@ -175,15 +175,31 @@ impl Key {
         algorithm: Algorithm,
         rng: &dyn rand::SecureRandom,
     ) -> Result<Self, error::Unspecified> {
-        Self::construct(algorithm, |buf| rng.fill(buf))
+        Self::construct(algorithm, algorithm.0.output_len, |buf| rng.fill(buf))
     }
 
-    fn construct<F>(algorithm: Algorithm, fill: F) -> Result<Self, error::Unspecified>
+    /// Generate an HMAC singing key using a custom length for the given
+    /// digest algorithm.
+    ///
+    /// The key will be `length` bytes long.
+    pub fn generate_with_length(
+        algorithm: Algorithm,
+        length: usize,
```

Comment:
Should this be `NonZeroUsize` at least? I don't think it makes sense to support zero-length keys.

Reply:
Done